spec/examples/schema_adapter_examples.rb

```diff
@@ -121,11 +127,9 @@
     it 'connects and resets' do
       subject.switch(schema1) do
         expect(connection.schema_search_path).to start_with %("#{schema1}")
-        expect(User.sequence_name).to eq "#{schema1}.#{User.table_name}_id_seq"
```

Comment:
Was there anything particularly wrong with this test? Shouldn't these assumptions still be valid?

Reply:
Ok, after starting to make some tests with my codebase and looking at the changes made, I now understand.
The previous changes tried to fix this issue, but ended up introducing other bugs while fixing this problem.
So, please correct me if i'm wrong, but i think we'll need to merge these changes, that will revert the state to "broken".
Release 2.9.1.

Then we can merge the changes i'm suggesting in #161 (assuming it looks ok), document the apartment model logic and release the 3.0.0.
Agree?

Reply:
I believe this PR solves the problem better, because `default_sequence_name` involves a trip to the database.

Reply:
Got it. I did a couple of tests and as long as we have the correct search path set, it should be ok and we should be getting the correct sequence.

Reply:
> I still think that having the logic on a concern that we include in our models would be a simpler solution than trying to manage this changes only when using postgres

I'd be willing to explore that but I'm not sure exactly what you have in mind. Do you mean something we `extend` into `ActiveRecord::Base`?

Reply:
Essentially something along these lines: https://github.com/rails-on-services/apartment/pull/169/files#diff-19a1d0c8bf6f60ce84cd210d933378fb77dd60a56232b05b0d2b89c1979c6331R8

I've adjusted the PR to follow your approach but with what i was suggesting